src/components/application_manager/include/application_manager/core_service.h

```diff
@@ -0,0 +1,257 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_CORE_SERVICE_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_CORE_SERVICE_H_
+
+#include <string>
+#include <vector>
+#include "application_manager/service.h"
+#include "application_manager/application.h"
+#include "policy/policy_types.h"
+
+namespace Json {
+class Value;
+}
+
+namespace application_manager {
+
+struct CommandParametersPermissions;
+
+/**
+ * @brief Class through which the plug-in can interact with the core
+ */
+class CoreService : public Service {
+ public:
+  /**
+   * @brief CoreService class destructor
+   */
+  explicit CoreService(ApplicationManager& application_manager);
+
+  /**
+   * @brief CoreService class destructor
+   */
+  virtual ~CoreService();
+
+  /**
+   * @brief Checks message permissions and cuts parameters according
+   * to policy table permissions
+   * @param msg message to cut disallowed parameters
+   * @return result according by mobile API
+   */
+  mobile_apis::Result::eType CheckPolicyPermissions(MessagePtr msg) FINAL;
+
+  /**
+   * Checks access to requested equipment of vehicle
+   * @param app_id id of application
+   * @param module type
+   * @param rpc name of rpc
+   * @param params parameters list
+   * @return return allowed if access exist,
+   * manual if need to send question to driver otherwise disallowed
+   */
+  TypeAccess CheckAccess(const ApplicationId& app_id,
+                         const std::string& module,
+                         const std::string& rpc,
+                         const std::vector<std::string>& params) FINAL;
+
+  /**
+   * Checks access to module for application
+   * @param app_id id of application
+   * @param module type
+   * @return true if module is allowed for application
+   */
+  bool CheckModule(const ApplicationId& app_id,
+                   const std::string& module) FINAL;
+
+  /**
+   * Sets access to functional group which contains given RPC for application
+   * @param app_id id of application
+   * @param module type
+   * @param allowed true if driver has given access
+   */
+  void SetAccess(const ApplicationId& app_id,
+                 const std::string& module,
+                 bool allowed) FINAL;
+
+  /**
+   * Resets access by group name for all applications
+   * @param group_name group name
+   */
+  void ResetAccess(const ApplicationId& app_id) FINAL;
+
+  /**
+   * Resets access by module type for all applications
+   * @param module type
+   */
+  void ResetAccess(const std::string& module) FINAL;
+
+  /**
+   * Gets device handler for device with certain ID
+   * @param device_id the ID of the connected device
+   * @return device handler if device with requested ID was found
+   */
+  uint32_t GetDeviceHandlerById(const std::string& device_id) FINAL;
+
+  /**
+   * Sets device as primary device
+   * @param dev_id ID device
+   */
+  void SetPrimaryDevice(const uint32_t dev_id) FINAL;
```

Comment:
@yang1070 
It used to be a drivers device, however now all devices considered as primary.